app/models/hyku_addons/csv_entry.rb

```diff
@@ -115,5 +113,56 @@ def path_to_file(file)
       return f if File.exist?(f)
       raise "File #{f} does not exist"
     end
+
+    ### Exporter overrides
+    def build_export_metadata
+      # Skip #make_round_trippable because it attempts to modify the original hyrax record
+      # which we don't need because we use the id of the hyrax record as the source_identifier
+      # make_round_trippable
+      self.parsed_metadata = {}
+      # We don't need a separate column for id because it is already in the source_identifer_field
+      # self.parsed_metadata['id'] = hyrax_record.id
+      parsed_metadata[self.class.source_identifier_field] = hyrax_record.id
+      parsed_metadata['model'] = hyrax_record.has_model.first
+      build_mapping_metadata
+      unless hyrax_record.is_a?(Collection)
+        parsed_metadata['file'] = hyrax_record.file_sets.map { |fs| filename(fs).to_s unless filename(fs).blank? }.compact.join('|')
+      end
+      build_json_metadata
+      parsed_metadata['visibility'] = hyrax_record.visibility
+      parsed_metadata['admin_set'] = hyrax_record.admin_set_id
+      parsed_metadata['collection'] = hyrax_record.member_of_collection_ids.join('|')
+      parsed_metadata
+    end
+
+    def build_mapping_metadata
+      export_mapping.each do |key, value|
+        method_name = value['from']&.first&.to_s || key.to_s
+        next unless hyrax_record.respond_to?(method_name)
+        data = hyrax_record.send(method_name)
+        if data.is_a?(ActiveTriples::Relation)
+          parsed_metadata[key] = data.map { |d| prepare_export_data(d) }.join('|').to_s unless value[:excluded]
+        else
+          parsed_metadata[key] = prepare_export_data(data)
+        end
+      end
+    end
+
+    def build_json_metadata
+      hyrax_record.class.json_fields.map(&:to_s).each do |field|
+        json_str = parsed_metadata.delete(field)
+        next unless json_str.present?
+        JSON.parse(json_str).each_with_index { |h, i| h.each { |k, v| parsed_metadata["#{k}_#{i + 1}"] = v } }
```

Comment:
This is one of these lines that, after 6 months of not touching the code, one hates himself because of his code hacking skills :).

Maybe a line of comment can help here.